mailbox/api/src/main/java/org/apache/james/mailbox/model/FetchGroup.java

```diff
@@ -0,0 +1,151 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.model;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Set;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Indicates the results fetched.
+ */
+public class FetchGroup {
+    /**
+     * For example: could have best performance when doing store and then
+     * forget. UIDs are always returned
+     */
+    public static final int NO_MASK = 0;
+    public static final int MINIMAL_MASK = 0x00;
+    public static final int MIME_DESCRIPTOR_MASK = 0x01;
+    public static final int HEADERS_MASK = 0x100;
+    public static final int FULL_CONTENT_MASK = 0x200;
+    public static final int BODY_CONTENT_MASK = 0x400;
+    public static final int MIME_HEADERS_MASK = 0x800;
+    public static final int MIME_CONTENT_MASK = 0x1000;
+
+    public static final FetchGroup MINIMAL = new FetchGroup(MINIMAL_MASK);
+    public static final FetchGroup HEADERS = new FetchGroup(HEADERS_MASK);
+    public static final FetchGroup FULL_CONTENT = new FetchGroup(FULL_CONTENT_MASK);
+    public static final FetchGroup BODY_CONTENT = new FetchGroup(BODY_CONTENT_MASK);
+
+    private final int content;
+    private final ImmutableSet<PartContentDescriptor> partContentDescriptors;
+
+    @VisibleForTesting
+    FetchGroup(int content) {
+        this(content, ImmutableSet.of());
+    }
+
+    @VisibleForTesting
+    FetchGroup(int content, ImmutableSet<PartContentDescriptor> partContentDescriptors) {
+        this.content = content;
+        this.partContentDescriptors = partContentDescriptors;
+    }
+
+    /**
+     * Contents to be fetched. Composed bitwise.
+     *
+     * @return bitwise description
```

Comment:
bitwise can be said of an operation but not of a value as far as I know.
I think it's rather a "bit field"

Reply:
Pfff will correct that javadoc. Tomorrow.